### PR TITLE
hw3_json_api

### PR DESCRIPTION
**Pull request by Fenix-125**
add meme-database-index JSON API

### CODE REVIEW

File: src/components/Catalog.js [hunk 1]
```diff
@@ -7,43 +7,52 @@ import Col from "react-bootstrap/Col";
...
+    componentDidMount() {
+        fetch(`/memes`).then((resp) => resp.json()).then(body => {
+            console.log(body);
+            this.setState({
+                info: body
+            })
+        });
+    }
+
+    render_cols(num, offset) {
```
Comment (pavlo-berezin) [inline src/components/Catalog.js:26]: camel case

File: src/components/Catalog.js [hunk 2]
```diff
@@ -7,43 +7,52 @@ import Col from "react-bootstrap/Col";
...
         let res = [];
-        for (let i = 0; i < num; ++i) {
-            res.push(<Col><MemeCard/></Col>);
+        for (let i = 0; this.state.info[i + offset * num] && i < num; ++i) {
```
Comment (pavlo-berezin) [inline src/components/Catalog.js:28]: consider using `slice` and `map` instead of this for.

File: src/components/MemeCard.js [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
+import ListGroupItem from "react-bootstrap/ListGroupItem";
 
 
 export default class MemeCard extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
     render() {
+        const info = {
```
Comment (pavlo-berezin) [inline src/components/MemeCard.js:13]: just `const info = this.props.info || {}`

File: src/components/MemeCard.js [hunk 4]
```diff
@@ -1,24 +1,32 @@
...
-                    </Card.Text>
+                    <Card.Title>{info.title}</Card.Title>
+                    <Card.Text>{info.explain}</Card.Text>
                 </Card.Body>
+
                 <ListGroup className="list-group-flush">
-                    <ListGroupItem>Cras justo odio</ListGroupItem>
-                    <ListGroupItem>Dapibus ac facilisis in</ListGroupItem>
+                    <ListGroupItem> </ListGroupItem>
+                    <ListGroupItem>info.category</ListGroupItem>
```
Comment (pavlo-berezin) [inline src/components/MemeCard.js:29]: You forgot `{` and `}` :)

**Comment by Fenix-125** [comment]
hw3

**Pull request merged**